File: blSystem-backend/src/main/java/com/blsystem/entity/Recipient.java
package com.blsystem.entity;

import com.blsystem.entity.enums.BloodGroup;
import com.blsystem.entity.enums.Gender;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Table(name = "recipients")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Recipient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Gender gender;
    private int age;
    private String address;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date benefitDate;
    private BloodGroup bloodGroup;
    private String contactNumber;
    private boolean isUrgent;
}

File: blSystem-frontend/src/app/model/Recipient.ts
import { BloodGroup } from "./enums/BloodGroup";
import { Gender } from "./enums/Gender";

export class Recipient {
    id!: number;
    name!: string;
    gender!: Gender;
    age!: number;
    address!: string;
    benefitDate!: Date;
    bloodGroup!: BloodGroup;
    contactNumber!: string
    urgent?: boolean 

    constructor(){}
}